src/Common/src/System/Net/ByteOrder.cs

```diff
@@ -0,0 +1,55 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Diagnostics;
+
+namespace System.Net
+{
+    internal static class ByteOrder
+    {
+        public static ushort HostToNetwork(this ushort host)
+        {
+#if BIGENDIAN
+            return host;
+#else
+            return unchecked((ushort)((host << 8) | (host >> 8)));
+#endif
+        }
+
+        public static uint HostToNetwork(this uint host)
+        {
+#if BIGENDIAN
+            return host;
+#else
+            return unchecked((uint)((host << 24) | ((host & 0xff00) << 8) | ((host & 0xff0000) >> 8) | ((host & 0xff000000) >> 24)));
+#endif
+        }
+
+        public static void HostToNetworkBytes(this ushort host, byte[] bytes, int index)
+        {
+            Debug.Assert(bytes != null);
+            Debug.Assert(sizeof(ushort) <= (bytes.Length - index));
+
+            bytes[index] = (byte)(host >> 8);
+            bytes[index + 1] = (byte)host;
```

Comment:
The bytes array is always in network byte order (i.e. big-endian). Since we're writing into it directly, it's always correct to write the MSB of the host value at the first byte, followed by the next-most-significant byte, and so on. If we were instead writing the value through a `ushort*` we'd need to twiddle the bytes appropriately.